ha_lineairdb.cc

```diff
@@ -624,8 +624,7 @@ int ha_lineairdb::rnd_next(uchar *buf) {
   memcpy(&buf[null_byte_cnt], &mask, 1);
 
   get_db()->EndTransaction(tx, [&](auto s) { status = s; });
-  p = inital_p;
-  free(p);
+  free((void *)init_p);
```

Comment:
[IMO] Do not use C-style cast! 
ref: https://docwiki.embarcadero.com/RADStudio/Sydney/ja/UCSC_--_C%2B%2B_%E3%82%B9%E3%82%BF%E3%82%A4%E3%83%AB%E3%81%AE%E3%82%AD%E3%83%A3%E3%82%B9%E3%83%88%E3%82%92%E4%BD%BF%E7%94%A8%EF%BC%88Use_C%2B%2B_Style_Casts%EF%BC%89